src/layer.rs

```diff
@@ -970,13 +970,19 @@ where
     }
 
     fn on_exit(&self, id: &span::Id, ctx: Context<'_, S>) {
+        let span = ctx.span(id).expect("Span not found, this is a bug");
+        let mut extensions = span.extensions_mut();
+
+        extensions
+            .get_mut::<OtelData>()
+            .expect("Missing otel data span extensions")
```

Comment:
Why is `expect()` the right call here? `on_close()` seems to avoid doing anything if `OtelData` is not present whereas `on_enter()` calls `expect()` like here. Does the disparity make sense?

Reply:
You're right, no need to panic. If the extension doesn't exist we should just do nothing. This can theoretically happen if someone uses a reloading layer.